integration-tests/profiler/profiler.spec.js

```diff
@@ -29,8 +29,7 @@ async function checkProfiles (agent, proc, timeout,
     }
   }, timeout, multiplicity)
 
-  await processExitPromise(proc, timeout, expectBadExit)
```

Comment:
How does this change the behaviour compared to existing code ?
It could improve error reporting and fail faster if `resultPromise` fails, but otherwise I have trouble finding cases were outcome would be different.